Adding documentation for new heartbeat accessors (#2250)
timescale/docs: api/_hyperfunctions/heartbeat_agg/num_gaps.md

---
api_name: num_gaps()
excerpt: Count the number of gaps between live ranges
topics: [hyperfunctions]
api:
  license: community
  type: function
  toolkit: true
  version:
    stable: 1.16.0
hyperfunction:
  family: state tracking
  type: accessor
  aggregates:
    - heartbeat_agg()
api_details:
  summary: |

>     Given a heartbeat aggregate, this will return the number of gaps between

Reviewer: ```suggestion
    Given a heartbeat aggregate, this returns the number of gaps between
```

>     the periods of liveness. Additionally if the aggregate is not live at the

Reviewer: ```suggestion
    the periods of liveness. Additionally, if the aggregate is not live at the
```

    start or end of its covered interval, these are also considered gaps.
  signatures:
    - language: sql
      code: |
        num_gaps(
            agg HEARTBEATAGG
        ) RETURNS BIGINT
  parameters:
    required:
      - name: agg
        type: HeartbeatAgg
        description: >
          A heartbeat aggregate to get the number of gaps from.
    returns:
      - column: num_gaps
        type: bigint
        description: >

>           The number of dead ranges in the aggregate.

Reviewer: ```suggestion
          The number of gaps in the aggregate.
```

Just to keep the wording consistent.

  examples:
    - description: >

>         Given a table called `liveness` containing weekly heartbeat aggregates in column `health` with timestamp column `date`, we can use the following to see how many times the system was down in a particular week.

Reviewer: ```suggestion
        Given a table called `liveness` containing weekly heartbeat aggregates in column `health` with timestamp column `date`, use this query to see how many times the system was down in a particular week:
```
